tensorflow/c/kernels.cc

```diff
@@ -279,4 +279,32 @@ TF_Tensor* TF_AllocateOutput(TF_OpKernelContext* context, int index,
     return nullptr;
   }
   return tf_tensor;
-}
\ No newline at end of file
+}
+
+TF_Tensor* TF_ForwardInputOrAllocateOutput(TF_OpKernelContext* context, 
+    int* candidate_input_indices, int num_input_indices, int output_index, 
```

Comment:
super nit. `num_input_indices` --> `num_candidate_input_indices`
Just to avoid a small chance of confusion that `num_input_indices` is number of all indices and not just candidate.